jena-base/src/main/java/org/apache/jena/atlas/lib/cache/CacheSimple.java

```diff
@@ -37,120 +37,81 @@
 
 public class CacheSimple<K,V> implements Cache<K,V>
 {
-    private final V[] values ; 
-    private final K[] keys ;
-    private final int size ;
-    private int currentSize = 0 ;
-    private BiConsumer<K,V> dropHandler = null ;
+	private Map<K,V> internalCache;
+	
```

Comment:
As you can see, I've added this change and also a test to demonstrate that `CacheSimple` now has the required fixed-size behavior.